src/services/refactors/inlineVariable.ts

```diff
@@ -0,0 +1,196 @@
+    }
+
+    // Try finding the declaration and nodes to replace via the reference token.
+    if (tryWithReferenceToken && isIdentifier(token)) {
+        const definition = checker.resolveName(token.text, token, SymbolFlags.Value, /*excludeGlobals*/ false);
+        if (definition?.declarations?.length !== 1) {
+            return { error: getLocaleSpecificMessage(Diagnostics.Variables_that_share_a_name_with_a_type_or_namespace_in_the_same_scope_cannot_be_inlined) };
+        }
+
+        const declaration = definition.declarations[0];
+        if (!isInitializedVariable(declaration) || !isVariableDeclarationInVariableStatement(declaration) || !isIdentifier(declaration.name)) {
+            return undefined;
+        }
+
+        const references = getReferenceNodes(declaration, checker, file);
+        return references && { references, declaration, replacement: declaration.initializer };
+    }
+
+    return { error: getLocaleSpecificMessage(Diagnostics.Could_not_find_variable_to_inline) };
+}
+
+function getReferenceNodes(declaration: InitializedVariableDeclaration, checker: TypeChecker, file: SourceFile): Identifier[] | undefined {
+    const references: Identifier[] = [];
+    const cannotInline = FindAllReferences.Core.eachSymbolReferenceInFile(declaration.name as Identifier, checker, file, ref => {
+        // Only inline if all references are reads. Else we might end up with an invalid scenario like:
+        // const y = x++ + 1 -> const y = 2++ + 1
+        if (FindAllReferences.isWriteAccessForReference(ref)) {
+            return true;
+        }
+
+        // typeof needs an identifier, so we can't inline a value in there.
+        if (isTypeQueryNode(ref.parent)) {
+            return true;
+        }
+
+        // Cannot inline recursive declarations (e.g. const foo = () => foo();)
+        if (textRangeContainsPositionInclusive(declaration, ref.pos)) {
+            return true;
+        }
+
+        references.push(ref);
+    });
+
+    return references.length === 0 || cannotInline ? undefined : references;
+}
+
+function getReplacementExpression(reference: Node, replacement: Expression): Expression {
+    // Logic from binaryOperandNeedsParentheses: "If the operand has lower precedence,
+    // then it needs to be parenthesized to preserve the intent of the expression.
+    // If the operand has higher precedence, then it does not need to be parenthesized."
+    //
+    // Note that binaryOperandNeedsParentheses has further logic when the precedences
+    // are equal, but for the purposes of this refactor we keep things simple and
+    // instead just check for special cases with needsParentheses.
+    const { parent } = reference;
+    if (isExpression(parent) && (getExpressionPrecedence(replacement) < getExpressionPrecedence(parent) || needsParentheses(parent))) {
+        return getSynthesizedDeepClone(factory.createParenthesizedExpression(replacement));

```

Comment:
Just reassign `replacement` with its own deep clone. You'll de-dupe some code and avoid re-cloning the parentheses every time.